deploy/index.js

```diff
@@ -0,0 +1,374 @@
+      `Response for latest request unexpected ${JSON.stringify(
+        latest,
+        null,
+        2
+      )}. `
+    );
+  } catch (e) {
+    console.log(e.message);
+    return;
+  }
+};
+
+const transformDownloadUrlToS3Format = (asset) => {
+  const downloadUrlBase = S3_BASE_URL + asset.name + "/";
+  // one way to deep copy
+  let ret = JSON.parse(JSON.stringify(asset));
+  ret.jar.download_url = downloadUrlBase + asset.jar.name;
+  ret.configuration.download_url = downloadUrlBase + asset.configuration.name;
+  ret.jvmRun.download_url = downloadUrlBase + asset.jvmRun.name;
+  return ret;
+};
+
+/**
+ * Request release data and write manifest file.
+ *
+ * returns true for success or false for failure
+ */
+const writeReleaseManifestAndLatest = async () => {
+  let response;
+  try {
+    response = await makeGithubApi();
+  } catch (e) {
+    console.error("Unable to request Github API data.");
+    console.error(e.message);
+    process.exit(-1);
+  }
+  console.log(JSON.stringify(response, null, 3));
+
+  const releases = filterNonFinalReleases(getDataFromResp(response));
+  const res = getReleaseData(sortReleasesByDatePublishDate(releases));
+  const filtered = res.filter(FilterOnlyCompleteReleases);
+  const transformedToS3 = filtered.map(transformDownloadUrlToS3Format);
+
+  writeFileSync(MANIFEST_FILE_NAME, JSON.stringify(transformedToS3, null, 2));
+
+  const latestTagName = await getLatestReleaseTagName();
+
+  let identifyLatestSuccess = false;
+
+  if (latestTagName) {
+    const latestIdx = filtered.findIndex(
+      (release) => release.name === latestTagName
+    );
+    if (latestIdx !== -1) {
+      writeFileSync(
+        LATEST_FILE_NAME,
+        JSON.stringify(transformedToS3[latestIdx], null, 2)
+      );
+      identifyLatestSuccess = true;
+    } else {
+      console.err(
+        `Could not find latest release version tag in releases ${latestTagName}.`
+      );
+    }
+  }
+  //  write the first version in the array !IMPORTANT - releases are sorted in order of release
+  if (!identifyLatestSuccess) {
+    if (filtered.len > 0) {
+      writeFileSync(
+        LATEST_FILE_NAME,
+        JSON.stringify(transformedToS3[0], null, 2)
+      );
+      identifyLatestSuccess = true;
+    } else {
+      console.error(
+        "All features were filtered, or not found in response. Unable to find latest release version."
+      );
+    }
+  }
+  let ret = {
+    fileWriteSuccess: identifyLatestSuccess,
+    releaseData: filtered,
+  };
+  return ret;
+};
+
+/**
+ * Upload manifest and JSON file
+ *
+ */
+const uploadManifestAndLatest = () => {
+  tryPut(MANIFEST_FILE_NAME, MIME_JSON);
+  tryPut(LATEST_FILE_NAME, MIME_JSON);
+};
+
+/**
+ * Upload Assets
+ *
+ */
+const uploadReleaseAssets = (assets) => {
+  for (let asset of assets) {
+    const folderName = asset.name;
+    // upload Jar
+    tryPutRemote(
+      asset.jar.download_url,
+      asset.jar.name,
+      folderName,
+      "application/java-archive"
```

Comment:
Create a constant for all MIME types to be consistent
```suggestion
      "MIME_JAR"
```

Reply:
changed